sensor/kubernetes/listener/resources/convert.go

```diff
@@ -350,8 +350,13 @@ func (w *deploymentWrap) populateImageMetadata(localImages set.StringSet, pods .
 			// If the ID already exists populate NotPullable, IsClusterLocal, and sync the registry
 			// and remote with the container runtime.
 			if image.GetId() != "" {
-				// Use the image ID from the pod's ContainerStatus.
-				image.NotPullable = !imageUtils.IsPullable(c.ImageID)
+				if c.ImageID != "" {
```

Comment:
I guess worst case is we attempt to scan an image which we cannot scan. This seems to just prevent the attempt, but the attempt isn't so bad

Reply:
And at this point an attempt would have already been made to scan the image as referenced in the deployment, this logic adjusts the deployment 'afterwards' from the pod data - it is likely (may be a bit of race) that the results from the previous scan attempt are already cached.